src/System.Reflection/tests/FieldInfo/FieldInfo_MethodTests.cs

```diff
@@ -13,69 +11,24 @@ namespace System.Reflection.Tests
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(expectedfieldValue));
-        }
-
-        //Verify GetValue method for non-static int FieldType
-        [Fact]
-        public void TestGetValue_NonStaticIntField()
-        {
-            String fieldname = "intFieldNonStatic";
-            int expectedfieldValue = 101;
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(expectedfieldValue));
-        }
-
-        //Verify GetValue method for static string FieldType
-        [Fact]
-        public void TestGetValue_StaticStrField()
+        // Verify GetValue method for fields
+        [Theory]
+        [InlineData("intFieldStatic", 100)]
+        [InlineData("intFieldNonStatic", 101)]
+        [InlineData("static_strField", "Static string field")]
+        [InlineData("nonstatic_strField", "NonStatic string field")]
+        public void TestGetValue(string fieldName, object expectedFieldValue)
         {
-            String fieldname = "static_strField";
-            String expectedfieldValue = "Static string field";
-            FieldInfo fi = GetField(fieldname);
             FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(expectedfieldValue));
-        }
-
-        //Verify GetValue method for non-static string FieldType
-        [Fact]
-        public void TestGetValue_NonStaticStrField()
-        {
-            String fieldname = "nonstatic_strField";
-            String expectedfieldValue = "NonStatic string field";
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(expectedfieldValue));
+            FieldInfo fi = GetAndTestFieldInfo(fieldName);
+            Assert.Equal(expectedFieldValue, fi.GetValue(myInstance));
         }
 
         //Verify GetValue method throws Exception when null is passed for non-static fields
         [Fact]
         public void TestGetValue_Exception1()
         {
-            String fieldname = "nonstatic_strField";
-            FieldInfo fi = GetField(fieldname);
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
+            FieldInfo fi = GetAndTestFieldInfo("nonstatic_strField");
 
             // In Win8p Instead of TargetException , generic Exception is thrown.
             // Refer http://msdn.microsoft.com/en-us/library/system.reflection.fieldinfo.getvalue.aspx
```

Comment:
Ah, actually, I was wrong.  I was thinking of TargetInvocationException, not TargetException.  It's correct here that the InnerException is null.  So it seems like we can update it to specify TargetException instead of Exception, and that should be sufficient.